### ISSUE TITLE
[Feature Request] Menu to compile 

### ISSUE THREAD

**Issue by gwwatkin** [latticesurgery-com/web-ui#93]
Issue Description
-----------------
Run example/upload circuit is not on the first page. Having that on the first page makes it much more encouraging for a user to actually try the compiler. 

Successful tools like [the Godbolt compiler explorer](https://godbolt.org/) have the can access the functionality on the main page.

Proposed Solution
-----------------
Two options
  1. Pass the circuit to the compile page and then run from there. Passing the circuit might be tricky. Options to explore are browser cache, fancy requests or an auxiliary lambda.
  2. Run the compilation on the main page, i.e the URL does not change after compiling. Ideally the compilation view would have it's own URL

Additional References
---------------------
https://stackoverflow.com/questions/52238637/react-router-how-to-pass-data-between-pages-in-react

**Comment by Keelando** [comment]
I agree. The Online Compiler tab and page can stay there, but we can add the compiler input on the home page too that redirects to the online compiler page along with the data, as suggested by that SO question
